My Project/clsTranslateWinForms.vb

```diff
@@ -1 +1,638 @@
+                End If
+            ElseIf TypeOf clsMenuItem Is ToolStripSplitButton Then
+                Dim clsTmpMenuItem As ToolStripSplitButton = DirectCast(clsMenuItem, ToolStripSplitButton)
+                If clsTmpMenuItem.HasDropDownItems Then
+                    GetDctComponentsFromMenuItems(clsTmpMenuItem.DropDownItems, dctComponents)
+                End If
+            ElseIf TypeOf clsMenuItem Is ToolStripDropDownButton Then
+                Dim clsTmpMenuItem As ToolStripDropDownButton = DirectCast(clsMenuItem, ToolStripDropDownButton)
+                If clsTmpMenuItem.HasDropDownItems Then
+                    GetDctComponentsFromMenuItems(clsTmpMenuItem.DropDownItems, dctComponents)
+                End If
+            End If
+
+        Next
+    End Sub
+
+    '''--------------------------------------------------------------------------------------------
+    ''' <summary>
+    '''     Attempts to translate all the text in <paramref name="dctComponents"/>
+    '''     to <paramref name="strLanguage"/>.
+    '''     Opens database <paramref name="strDataSource"/> and reads in all translations for the 
+    '''     <paramref name="strControlName"/> control for target language <paramref name="strLanguage"/>.
+    '''     For each translation in the database, attempts to find the corresponding component in 
+    '''     <paramref name="dctComponents"/>. If found, then it translates the text to the target 
+    '''     language. If a component has a tool tip, then it also translates the tool tip.
+    ''' </summary>
+    '''
+    ''' <param name="dctComponents">    [in,out] The dictionary of translatable components. </param>
+    ''' <param name="strControlName">   The name of the form or menu used to populate the dictionary. </param>
+    ''' <param name="strDataSource">    The path of the SQLite '.db' file that contains the
+    '''                                 translation database. </param>
+    ''' <param name="strLanguage">      The language code to translate to (e.g. 'fr' for French). </param>
+    '''
+    ''' <returns>
+    '''     If an exception is thrown, then returns the exception text; else returns 'Nothing'.
+    ''' </returns>
+    '''--------------------------------------------------------------------------------------------
+    Private Shared Function TranslateDctComponents(ByRef dctComponents As Dictionary(Of String, Component),
+                                                   strControlName As String,
+                                                   strDataSource As String,
+                                                   strLanguage As String) As String
+        'Create a list of all the tool tip objects associated with this (sub)dialog
+        'Note: Normally, a (sub)dialog wil only have a single tool tip object. This stores the 
+        '      tool tips for all the components in the (sub)dialog.
+        Dim lstToolTips = New List(Of ToolTip)
```

Comment:
```suggestion
        Dim lstToolTips As New List(Of ToolTip)
```

Reply:
This is the simplification suggested by Visual Studio:

![image](https://user-images.githubusercontent.com/57253949/147076027-df9d0c0d-dfcb-4ed6-a6d9-b93653793bfe.png)

Do you have strong objections if I leave it as is?

Reply:
Ooh I guess it suggested that because;

`Dim lstToolTips = New List(Of ToolTip)` because you had `Dim lstToolTips As List(Of ToolTip) = New List(Of ToolTip)` instead of `Dim lstToolTips As New List(Of ToolTip)`.

I have no strong objection, they may have changed the syntax standards of defining new objects and improved the compiler to detect its type based on this kind of declaration. Back in the days of older visual studio, objects defined that way didn't have their properties recognized during runtime, the compiler would do implicit conversions when accessing its properties during runtime.

Reply:
@Patowhiz I just re-read my comment above, and I was wrong. Visual Studio actually suggests the same as you. Therefore I will implement your original suggestion. Sorry for the confusion.